scripts/konflux_resolve.py

```
@@ -0,0 +1,1179 @@
+        self.index_url = index_url.rstrip("/") + "/"
+        self.python_version = python_version
+        self.platforms = list(platforms)
+        self._parser = SimpleIndexParser()
+        self._known_packages: set[str] = set()
+        self._packages: dict[str, dict[str, dict[str, tuple[str, str]]]] = {}
+
+    def _fetch_url(self, url: str) -> str:
+        last_exc: Exception | None = None
+        for attempt in range(3):
+            try:
+                with urllib.request.urlopen(url, timeout=30) as resp:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)